remotes/docker/fetcher.go

```diff
@@ -210,5 +214,30 @@ func (r dockerFetcher) open(ctx context.Context, req *request, mediatype string,
 		}
 	}
 
-	return resp.Body, nil
+	body := resp.Body
+	encoding := strings.Split(resp.Header.Get("Content-Encoding"), ", ")
+	for i := len(encoding) - 1; i >= 0; i-- {
+		algorithm := encoding[i]
+		switch algorithm {
+		case "zstd":
+			r, err := zstd.NewReader(body)
+			if err != nil {
+				return nil, err
+			}
+			body = r.IOReadCloser()
```

Comment:
Needs to break the loop here?

Reply:
no, when response was encoded using multiple algorithm, we need to apply them all, in reverse order
Probably never happens, but as suggested by @thaJeztah, better strictly conform with HTTP RFC to avoid future issues

Reply:
Thanks! Yes, I think it's good to at least have the code in this PR.

Whether or not all things should be enabled by default, or if this should be a configurable option (which compressions to support), can still be discussed